vertx-template-engines/vertx-web-templ-pebble/src/main/java/io/vertx/ext/web/templ/impl/PebbleTemplateEngineImpl.java

```diff
@@ -43,61 +38,48 @@
  */
 public class PebbleTemplateEngineImpl extends CachingTemplateEngine<PebbleTemplate> implements PebbleTemplateEngine {
 
-	private final PebbleEngine pebbleEngine;
-	private final FileLoader fileLoader;
-	private final ClasspathLoader classpathLoader;
+  private final PebbleEngine pebbleEngine;
+  private final Loader<String> loader;
 
-	public PebbleTemplateEngineImpl() {
-		super(DEFAULT_TEMPLATE_EXTENSION, DEFAULT_MAX_CACHE_SIZE);
+  public PebbleTemplateEngineImpl(Vertx vertx) {
+    super(DEFAULT_TEMPLATE_EXTENSION, DEFAULT_MAX_CACHE_SIZE);
+    loader = new PebbleVertxLoader(vertx);
+    loader.setSuffix(DEFAULT_TEMPLATE_EXTENSION);
```

Comment:
That is true because the resolution of the path is done at the `CachingTemplateEngine` which will add the correct extension. I can remove these call since they are NO-OPs but i can't remove the method from the loader since it is declared on the Pebble loader interface